test/controllers/friendships_controller_test.rb

```diff
@@ -0,0 +1,54 @@
+require 'test_helper'
+class FriendshipsControllerTest < ActionDispatch::IntegrationTest
+
+  # setup user for friend to friends
+  def setup
+    @user1 = users(:michael)
+    @user2 = users(:archer)
+    @friendship = friendships(:one)
+  end
+
+  #test for send friend request to other user
+  test 'Sends the request' do
+    get user_session_path
+    post user_session_path, params: { user: {
+                                               email: 'mugoiri@example.com',
+                                               password: 'password'
+                                            }
+                                    }
```

Comment:
Can we use devise sign in?

Reply:
yes ma'am

Reply:
in this test cases using devise sign in

Reply:
You are using the exact test case as written in the book.

You should use devise helpers in tests if you are using it in app.

Reply:
yes ma'am I follow the book but surly read devise helpers in tests